### ISSUE TITLE
privacy settings are not persisting if firefox not restarted after ublock install

### ISSUE THREAD

**Issue by 2vek** [gorhill/uBlock#343]
I installed ublock 0.9.9.0-rc.0 over ublock 0.9.8.2 (installed via AMO) on Firefox 38.0.5.

Both privacy settings (i.e. disable prefetching and hyperlink auditing) were unchecked.
So, I checked both settings and closed dashboard.

Opened dashboard again after few minutes. Both privacy setting were unchecked. So, I checked them again and closed the dashboard. Opened dashboard again, both privacy settings were still unchecked.

Tried above with other settings like 'i am advanced user', 'color-blind friendly'
etc and these settings were persisting as they should.

I restarted Firefox and problem vanished. Now privacy settings are persisting like other settings.

**Comment by gorhill** [comment]
> I installed ublock 0.9.9.0-rc.0 over ublock 0.9.8.2

Can you describe how exactly you did this?

**Comment by gorhill** [comment]
I installed 0.9.8.2, quit Firefox 38, launched Firefox, drag and dropped 0.9.9.0-rc.0 in Firefox, and the checkboxes worked as expected. I don't know how to recreate your issue. From the description, it is as if the code from 0.9.8.2 was still in effect, while the dashboard code was that of 0.9.9.0, i.e. the core didn't know about these new privacy settings while the dashboard knew about them.

**Comment by 2vek** [comment]
> Can you describe how exactly you did this?

I installed ublock few week ago from AMO. So, I had ublock 0.9.8.2 installed before installing 0.9.9.0-rc.0.
I went to [ublock release page](https://github.com/gorhill/uBlock/releases). Clicked on `uBlock0.firefox.xpi`.
Firefox shows a small popup below address bar asking to allow github.com to install software. Click on 'allow'. A download progress bar is shown. A giant dialog box shown with title 'Software Installation', click on 'Install now'. Then 'successfully installed' message is shown below address bar.

After that, I went to `about:addons` page to access ublock preferences. Then clicked to dashboard button. (I have ublock icons hidden from toolba). 

> I installed 0.9.8.2, quit Firefox 38, launched Firefox, drag and dropped 0.9.9.0-rc.0 in Firefox

I was able to reproduce it again, by doing just that. It's reproducible on my system every time.

> it is as if the core code from 0.9.8.2 was still in effect, while the dashboard code was that of 0.9.9.0

That's what I thought, which is why I restarted(close all windows then open) Firefox.

Also I noticed few things when I installed 0.9.9.0-rc.0 again to reproduce bug.
* ublock's about page  and ublock popup ui show version 0.9.8.2, `about:addons` show version 0.9.9.0
* new feature added in logger didn't work.
* after restarting Firefox, privacy settings checkboxes are checked by default. Versions numbers are cosistent everywhere. Able to use new features of logger.

**Comment by gorhill** [comment]
> I have ublock icons hidden from toolbar

Is it in the menu? Or removed from toolbar and menu?

**Comment by 2vek** [comment]
Icons were hidden from both toolbar and hamburger menu. I tried putting ublock icon back in toolbar and repeated the steps but bug still appears.

**Comment by MiraziHeket** [comment]
Is that same like here: https://github.com/gorhill/uBlock/issues/321?

**Comment by 2vek** [comment]
I have launched Firefox at least 10 times since creating this issue. None of my settings or filters are reset or lost. 

It's just that, very first time when I installed latest ublock, privacy setting were unchecked and new features were inaccessible. After I restarted firefox, every thing is working as expected. No problem since then.

@MiraziHeket If your settings resets every time then It is different bug.

**Comment by 2vek** [comment]
So, I updated ublock from 0.9.9.0 to 0.9.9.1, and bug appears to be gone.

To be sure, without restarting firefox I checked if new by ublock is autoreloaded on update.
* by checking ublock version on `about:addons` page, ublock popup ui, and ublock dashboard about page. They were consistent
* I checked for new enhancements in ublock https://github.com/gorhill/uBlock/issues/346 and https://github.com/gorhill/uBlock/issues/345 . They worked.

That's how I concluded that bug is fixed.

I do not wish to waste developer time on this bug, as it was not a major bug and is solved.
